lib/elixir/lib/kernel/typespec.ex

```diff
@@ -883,7 +887,9 @@ defmodule Kernel.Typespec do
 
   # Handle remote calls
   defp typespec({{:., meta, [remote, name]}, _, args} = orig, vars, caller) do
-    remote = Macro.expand remote, caller
+    # We set a function name to avoid tracking
+    # aliases in typespecs as compile time dependencies.
+    remote = Macro.expand remote, %{caller | function: {:typespec, 0}}
```

Comment:
There is some existing code that uses this approach here:
https://github.com/antipax/elixir/blob/99a911aee81ce634f0725e274b2ea7ed131799b1/lib/elixir/lib/kernel/typespec.ex#L853

I think it's still a good idea to track the reference as runtime, so personally I'd be against disabling the lexical tracker, unless we want to consider types as existing outside of normal code. I don't think we need to change the env function name since for tracking purposes it's only used in terms of presence vs. nil to determine whether it's compile or runtime.